keps/sig-testing/2539-continuously-deploy-k8s-prow/README.md

```diff
@@ -0,0 +1,149 @@
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This document proposes to change deployment of k8s prow from manual to be automated continuously.
+
+## Motivation
+
+Currently, deploying k8s prow consists of following steps:
+
+1. Updates made in prow are released as container images.
+1. Automated process creates PRs updating the prow images tags.
+1. Oncall inspects prow logs to make sure prow is safe to upgrade.
+1. Oncall inspects PRs created in step #2, approve them, and post message on Slack.
+1. Oncall waits until #4 and a postsubmit job applying the changes on prow cluster.
+1. Oncall waits until #5 is done, do several manual inspections to make sure prow works.
+
+This is a very time consuming process (Roughly 3 hours per week for oncall), especially the context switching between the waits makes oncall hard to focus on their day job. Thus it’s desired to streamline the process so that we can save time.
+
+Historically, the biggest reasons why the manual processes were introduced are:
+
+- Errors in prow were not easily discoverable, most of time were reported by prow users, which was bad.
+
+This problem has been largely solved by the introduction of prow monitoring + alerting by grafana, prometheus, and prometheus alertmanager stack. And based on our experience in the past quarter, we haven’t had a single case where prow errors were discovered by humans earlier than by prow alerts. This fact gives us reasonable confidence to proceed with continuous delivery, and the following assumptions should hold:
+
+- Prow is stable as long as there is no alert. (Indicate that no need to inspect prow logs before bumping)
+- Errors caused by prow upgrades are discovered by prow alerts in a timely manner.
+
+
+### Goals
+
+The proposed change switches from daily manual deployment to hourly automated deployment.
+
+#### Prow Users
+
+Shouldn’t see any change, prow breakage should be discovered by prow monitoring system and rollback will be performed. The chance of prow being break is almost identical to what we have today(Assume there are not more than a single breaking change every day).
+
+#### Prow Oncall
+
+- What’s Not Changed
+  - React to prow alerts and take actions.
+- What’s Changed
+  - No more manual inspecting prow healthiness.
+  - No more manual lgtm/approve/retest autobump PRs.
+  - No more manual Slack posting.
+
+
+### Non-Goals
+
+Change how prow is released.
+
+
+## Proposal
+
+Prow autobump PRs are automatically merged every hour, only on working hours of working days.
```

Comment:
We discussed in sig testing meeting starting at a much longer interval given that some jobs don't complete for O(2h), and having a slower update rate may make for easier troubleshooting

Reply:
I'll start with 3 hours for now.